client/rpmtrans.c

```diff
@@ -480,9 +465,51 @@ TDNFTransAddInstallPkg(
                   fp,
                   pszFilePath,
                   &rpmHeader);
-    //If not checking gpg sigs, ignore signature errors
-    if(!nGPGSigCheck && (dwError == RPMRC_NOTTRUSTED || dwError == RPMRC_NOKEY))
+
+    Fclose(fp);
+    fp = NULL;
+
+    if(nGPGSigCheck && (dwError == RPMRC_NOTTRUSTED || dwError == RPMRC_NOKEY))
     {
+        dwError = TDNFGetGPGSignatureCheck(pTdnf, pszRepoName, &nGPGSigCheck, &pszUrlGPGKey);
+        BAIL_ON_TDNF_ERROR(dwError);
+
+        printf("importing key from %s\n", pszUrlGPGKey);
+        dwError = TDNFYesOrNo(pTdnf->pArgs, "Is this ok [y/N]: ", &nAnswer);
+        BAIL_ON_TDNF_ERROR(dwError);
+
+        if(nAnswer)
+        {
+            pKeyring = rpmtsGetKeyring(pTS->pTS, 0);
+
+            dwError = TDNFImportGPGKey(pTS->pTS, pszUrlGPGKey);
+            BAIL_ON_TDNF_ERROR(dwError);
+
+            dwError = TDNFGPGCheck(pKeyring, pszUrlGPGKey, pszFilePath);
+            BAIL_ON_TDNF_ERROR(dwError);
+
+            fp = Fopen (pszFilePath, "r.ufdio");
+            if(!fp)
+            {
+                dwError = errno;
+                BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
+            }
+
+            dwError = rpmReadPackageFile(
+                          pTS->pTS,
+                          fp,
+                          pszFilePath,
+                          &rpmHeader);
+
+            BAIL_ON_TDNF_RPM_ERROR(dwError);
+
+            Fclose(fp);
+            fp = NULL;
```

Comment:
is this assignment really needed? There is one more above, I feel it won't make a difference

Reply:
After closing it should be set to NULL, so it's not closed on cleanup.

Reply:
Agreed, ignore my comment about null assignment